### PR TITLE
Adding PSR-3, PSR-18, PSR-15 as provided

### PR DESCRIPTION
**Pull request by challgren**
Some more implementations that are provided by CakePHP but not defined in the composer.json file.

### CODE REVIEW

**Comment by challgren** [comment]
I think I got all the packages fixed up.

File: src/Core/composer.json [hunk 1]
```diff
@@ -25,6 +25,9 @@
         "php": ">=7.4.0",
         "cakephp/utility": "^4.0"
     },
+    "provide": {
+        "psr/http-server-handler-implementation": "^1.0"
```
Comment (ADmad) [inline src/Core/composer.json]: This doesn't seem right.

Comment (ADmad) [reply src/Core/composer.json]: Ah, it's cause `HttpApplicationInterface` in this package extends `RequestHandlerInterface`. But having an interface is not the same as providing an implementation.

Comment (challgren) [reply src/Core/composer.json]: Yeah I mainly looked for any implementations and fixed those packages

Comment (ADmad) [reply src/Core/composer.json]: I believe an implementation requires a class, not just an interface.

**Comment by markstory** [comment]
Thank you 🎉

**Pull request merged**